src/main/java/de/dennisguse/opentracks/viewmodels/AggregatedStatisticsModel.java

```diff
@@ -21,18 +20,26 @@ public AggregatedStatisticsModel(@NonNull Application application) {
         super(application);
     }
 
-    public LiveData<AggregatedStatistics> getAggregatedStats(@Nullable List<Track.Id> trackIds) {
+    public LiveData<AggregatedStatistics> getAggregatedStats(Track.Selection selection) {
         if (aggregatedStats == null) {
             aggregatedStats = new MutableLiveData<>();
-            loadAggregatedStats(trackIds);
+            loadAggregatedStats(selection);
         }
         return aggregatedStats;
     }
 
-    private void loadAggregatedStats(@Nullable List<Track.Id> trackIds) {
+    public void updateSelection(Track.Selection selection) {
+        loadAggregatedStats(selection);
+    }
+
+    public void clearSelection() {
+        loadAggregatedStats(Track.Selection.create());
```

Comment:
Not yet sure, if Selection should be an inner class of Track.
It somehow feels like separate class that is used for selecting tracks.

Reply:
I think so, because it's a track's selection. If you remove Track class then you have to remove Track.Selection one.

Reply:
That's true, but not every Track has a selection and the selection is only relevant for the filtering.
So for me, the selection is some structure that helps us to find the correct tracks, but afterwards the tracks have nothing to with it.
And it somehow only is used by one feature (at the moment) while the tracks are used almost everywhere.